### PR TITLE
Mitigated FP-precision problem in label to flow conversion code and double input file detection bug

### PR DESCRIPTION
**Pull request by koerner-axs**
Replaced imprecise calculations of the form log(1+x) with specialized functions from Torch and NumPy.

The previous implementation caused numerical instability when very long cells are converted to flows.
I observe the diffusion process implementation to be quite robust, however going from mass concentration to log mass concentration + 1 is introducing instability. This is because to potentially very small mass concentrations (maybe 10e-50 in my particular case) we add 1.0 and only then take the logarithm. Even for float64 this gap between the exponents is too large.

To resolve this problem, a number of libraries provide specialized functions to perform the log(1+x) operation using a single call to a function commonly named log1p.

In general, I am skeptical about the need to compute the log mass concentrations. Because the flow field is normalized afterwards anyways, dropping the logarithm seemed to work fine for me.

### CODE REVIEW

**Comment by koerner-axs** [comment]
This image shows the dFdy, that it computed for one of my labeled training images. In the top right, you can see what issues I am trying to circumvent with this pull req.
example-precision-error

**Comment by koerner-axs** [comment]
Last commit also takes care of this issue: https://github.com/MouseLand/cellpose/issues/626

**Comment by carsen-stringer** [comment]
thanks can you remove the second fix? I will put testing the log1p on my to-do list (want to test the full training with the original cellpose training set)

**Comment by koerner-axs** [comment]
I have now reset my fork to the log1p commit.
I really should have split these two issues up.

**Comment by carsen-stringer** [comment]
thanks @koerner-axs we tried using the logexp function in pytorch, but found removing the log computation altogether worked best. this will be available in the new release
